src/app.js

```diff
@@ -557,6 +572,18 @@ var run = function () {
         files.addReadOnly(url, content)
         cb(null, content)
       })
+    } else if ((match = /^(ipfs:\/\/?.+)/.exec(url))) {
+      $('#output').append($('<div/>').append($('<pre/>').text('Loading ' + url + ' ...')))
+      handleIPFS(match[1], function (err, content) {
+        if (err) {
+          cb('Unable to import "' + url + '": ' + err)
```

Comment:
There is lots of code duplication between the three remote imports. Can you reduce this to something like
```js
var importers = [{match:  /^(https?:\/\/)?(www.)?github.com\/([^/]*\/[^/]*)\/(.*)/, importer = handleGithubImport}, {...}]
for (...) { ...}
```
?